src/python/pants/backend/python/tasks2/pex_build_util.py

```diff
@@ -119,6 +128,68 @@ def dump_requirements(builder, interpreter, req_libs, log, platforms=None):
       locations.add(dist.location)
 
 
+def build_python_distribution_from_target(target, workdir):
+  # Create a temporary working directory, pydistworkdir
+  pydist_workdir = os.path.join(workdir, 'pydistworkdir')
+  safe_mkdir(pydist_workdir)
+  pex_name = "%s.pex" % target.name
+  path_to_target = os.path.dirname(target.address.rel_path)
+
+  # disable cache to ensure reproducible runs
+  args = ['--disable-cache', path_to_target, '-o', os.path.join(pydist_workdir, pex_name)]
+  try:
+    pex_main.main(args=args)
```

Comment:
Would the api be able to be as high level as:
```
resolvables = [Resolvable.get(arg, resolver_option_builder) for arg in args]
resolveds = resolve_multi(resolvables)
```
Is this what you had in mind? As in , skipping the pex packaging all together and these resolved distributions directly instead of unzipping?

Reply:
no, not quite. I'd take a look at the pants `setup-py` task here: https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/python/tasks2/setup_py.py

to see how that works and how it handles products. this task essentially generates `setup.py` files for a given targets and can execute against the generated `setup.py` to produce e.g. an sdist or bdist (e.g. `./pants setup-py ... --run="bdist_wheel"` will net a .whl in the dist dir). you might even be able to directly recycle parts of this as part of a more first class product pipeline vs doing the wheel build directly inline. this would also theoretically net you "free" caching.

Reply:
✅